steamclog/src/main/java/com/steamclock/steamclog/ThrowableFilter.kt

```diff
@@ -0,0 +1,11 @@
+package com.steamclock.steamclog
+
+/**
+ * ThrowableFilter functional interface (ie. should only contain a single method).
+ * Allows the application to intercept Throwables when an error occurs and decide if the
+ * Throwable should be blocked from being logged as an error by the crash reporting
+ * destination.
+ */
+fun interface ThrowableFilter {
```

Comment:
Because the function name is hidden when implementing the interface in a lambda, we'll probably need to make sure this interface name is really clear instead. Maybe ThrowableBlocker would make the usage more clear?

```kotlin
clog.throwableFilter = ThrowableBlocker { throwable ->
            when (throwable) {
                is BlockedException1 -> {
                    true
                }
                is BlockedException2 -> {
                    true
                }
                else -> {
                    false
                }
            }
        }
```

I'm not sure if that's actually any more readable though. @munl @jenncoop?

Reply:
Yeah, I think so - `ThrowableBlocker` seems the closest fit to what the fun interface is intended for 👍